### PR TITLE
wip: non-trivial gas savings by packing index into reserves slot

### PR DESCRIPTION
**Pull request by OliverNChalk**
Question: Should we leave the reserves alone and just store the index in the same slot as the `bool lock`  for re-entrancy guard? You could also turn this on its head and ask can we get away wit ha 24bit timestamp (though this seems much dodgier unless you `/10` each timestamp).

### CODE REVIEW

**Review by xenide — APPROVED** [review]
Reviewed the failing test, can confirm that it is due to the precision change. You can increase the tolerance for the test to pass.

File: src/asset-management/AaveManager.sol [hunk 1]
```diff
@@ -51,18 +51,18 @@ contract AaveManager is IAssetManager, Owned(msg.sender), ReentrancyGuard {
...
-    function _getBalance(IAssetManagedPair aOwner, address aToken) private view returns (uint112 rTokenBalance) {
+    function _getBalance(IAssetManagedPair aOwner, address aToken) private view returns (uint104 rTokenBalance) {
         address lAaveToken = _getATokenAddress(aToken);
         uint256 lTotalShares = totalShares[lAaveToken];
         if (lTotalShares == 0) {
             return 0;
         }
         rTokenBalance =
-            uint112(shares[aOwner][aToken] * IERC20(lAaveToken).balanceOf(address(this)) / totalShares[lAaveToken]);
+            uint104(shares[aOwner][aToken] * IERC20(lAaveToken).balanceOf(address(this)) / totalShares[lAaveToken]);
```
Comment (OliverNChalk) [inline src/asset-management/AaveManager.sol:65]: Overflow?

Comment (xenide) [reply src/asset-management/AaveManager.sol:65]: I think we should be good cuz `balanceOf` returns a uint256 which the `uint104` will be implicitly cast into during the operation

Comment (OliverNChalk) [reply src/asset-management/AaveManager.sol:65]: There is no overflow checking when you cast uint256 to uint104 in solidity pretty sure?

Comment (xenide) [reply src/asset-management/AaveManager.sol:65]: that's correct. No overflow check. But with the amounts we are managing, it's unlikely that we'd reach that limit. However, if we're still worried, we can use openzeppelin's `SafeCast::toUint104()` to ensure that it is indeed within range

Comment (OliverNChalk) [reply src/asset-management/AaveManager.sol:65]: Yeah, I'd probably prefer a safeCast here even though its extremely unlikely its still possible

Comment (xenide) [reply src/asset-management/AaveManager.sol:65]: okay created #140 to make sure we don't miss it. 

Resolving for now

**Pull request merged**